python/tank/platform/application.py

```diff
@@ -117,7 +120,40 @@ def engine(self):
         The engine that this app is connected to.
         """
         return self.__engine
-        
+
+    @property
```

Comment:
i changed things around quite a bit so there is a lot less stuff in application and framework now. 

It's basically just creating the self._log, which is different depending for app/fw classes, and then the self.log property which has different doc strings for the classes. I think it gets confusing if we try to make it more generic, pls shout if you disagree!